### PR TITLE
add proxy for lspd htlc notifications

### PR DESCRIPTION
**Pull request by JssDWt**
Add a proxy to be able to subscribe to htlc notifications from lspd through Breez server.

### CODE REVIEW

File: lsp/lsp.go [hunk 1]
```diff
@@ -245,9 +254,32 @@ func (s *Server) CheckChannels(ctx context.Context, in *breez.CheckChannelsReque
...
 	clientCtx := metadata.AppendToOutgoingContext(context.Background(), "authorization", "Bearer "+lsp.Token)
-	reply, err := lspdClient.CheckChannels(clientCtx, &lspdrpc.Encrypted{Data: in.Blob})
+	reply, err := lspdClient.channelOpenerClient.CheckChannels(clientCtx, &lspdrpc.Encrypted{Data: in.Blob})
 	if err != nil {
 		return nil, err
 	}
 	return &breez.CheckChannelsReply{Blob: reply.Data}, nil
 }
+
+func (s *Server) RegisterHtlcNotification(
```
Comment (roeierez) [inline lsp/lsp.go:264]: NIT: Perhaps a name that exposes higher level of abstraction? `RegisterIncomingPaymentNotification` or just `RegisterPaymentNotification`
Using htlc might make the user think she should get a notification for every htlc.

**Review by roeierez — COMMENTED** [review]
Other than a small Nit LGTM

**Comment by JssDWt** [comment]
renamed to `RegisterPaymentNotification`
Also I forgot to register that subserver to the grpc server. That's added now.

**Review by roeierez — APPROVED** [review]
LGTM

**Pull request merged**